lib/phoenix_datatables/query.ex

```diff
@@ -17,21 +17,21 @@ defmodule PhoenixDatatables.Query do
   For some queries, `:columns` need to be passed - see documentation for `PhoenixDatatables.execute`
   for details.
   """
-  def sort(queryable, params, sortable \\ nil)
-  def sort(queryable, %Params{order: orders} = params, sortable) when is_list(sortable) do
+  def sort(queryable, params, options \\ nil)
+  def sort(queryable, %Params{order: orders} = params, [columns: columns] = options) when is_list(options) do
```

Comment:
Matching like this is unsound; matching a Keyword list is not like matching a map. Both the number of and order of the items has to match. This pattern would only match the case that there is only one keyword in the list and it is columns. You could keep this approach if you have `execute` call Map.new on the options, and then match them as a map here.

Reply:
Is it acceptable to convert `options` to a map everywhere? So `execute` would take a `Map.t` rather than converting a `Keyword.t`. Sorry for the delay.

Reply:
No, keyword is the accepted idiom for options in Elixir because it has syntax sugar when it is the last arg in a function call. Conversion to map is sometimes done internally for matching support.  This is how Phoenix json render functions work, for example. Thinking that through though, since sort is technically a public function it might be better to have `sort` also take a keyword list, and then either convert to map and re-dispatch to an internal function or check/assign each arg that it uses.

Reply:
Ok, how about this approach?